### PR TITLE
Add MQL context

### PR DESCRIPTION
**Pull request by adamrneary**
This enables us to create a valid MQL Context. It does not yet use that MQL Context for any fetching, but it's a start. 

Screen Shot 2021-03-05 at 10 07 04 AM

We _should_ be able to swap out the version we are using in the product for this version seamlessly. Then we can begin to add in the other features we require for querying the MQL Server.

### CODE REVIEW

File: context/MqlContext/MqlContextProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  captureException?: (e: CombinedError) => void;
+  children: ReactNode;
+};
+
+/*
+  This thin wrapper allows us to use an urql Provider to provide access to our API
+  for the sake of interacting with MQL Server and Model Key selection
+*/
+function MqlContextProvider({ getToken, children, captureException }: Props) {
+  console.log("a", buildApiUrqlClient(getToken));
```
Comment (jjrajani) [inline context/MqlContext/MqlContextProvider.tsx:24]: nit: log

File: context/MqlContext/MqlContext.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { Client } from "urql";
+import { createContext } from "react";
+
+export type MqlContextType = {
+  /*
+    The MQL Server URL can be overridden by a user in order to access a development
+    server or a test server as needed. This value represents the currently selected 
+    URL for the user and is not optional.
+  */
+  mqlServerUrl?: string | null;
```
Comment (jjrajani) [inline context/MqlContext/MqlContext.ts:10]: Is not optional? Do we want the `?` then?  Or would we prefer a type error to throw?

File: utils/buildAuthExchange.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { makeOperation } from "urql";
+import { authExchange } from "@urql/exchange-auth";
+
+function buildGetAuth(getToken: () => Promise<string>) {
+  const getAuth = async ({ authState }) => {
+    // TODO: We could start parsing and evaluating this token to avoid having to fetch it each time
+    const token = await getToken();
+
+    if (!authState) {
```
Comment (jjrajani) [inline utils/buildAuthExchange.ts:9]: Just asking for more context.  I'm confused about what this does if there is authState,  just return null?  Would that mean no token is provided?

**Review by jjrajani — APPROVED** [review]

File: context/MqlContext/MqlContext.ts (hunk 2 above)
Comment (adamrneary) [reply context/MqlContext/MqlContext.ts:10]: This is optional in the sense that on initial load it is not yet retrieved from the server.

File: utils/buildAuthExchange.ts (hunk 3 above)
Comment (adamrneary) [reply utils/buildAuthExchange.ts:9]: I am trying to wrap my head around it, also. This came directly from one of the examples. I need to dig into it more to better understand it myself.

**Pull request merged**